### PR TITLE
adding sort by usage feature with revised changes

### PR DESCRIPTION
**Pull request by lahiru823**

### CODE REVIEW

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag [hunk 1]
```diff
@@ -93,7 +93,46 @@ require('/modules/store.js').exec(function (ctx) {
                 }
```
Comment (dinusha-dilrukshi) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: Comments given in https://github.com/wso2/product-app-manager/pull/78. apply to here as well.

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag [hunk 2]
```diff
@@ -93,7 +93,46 @@ require('/modules/store.js').exec(function (ctx) {
...
+        }
+        else if (paging.sort == 'usage') {
+            server.privileged(function () {
+                var config = require('/config/store.json');
+                var userName = user ? user.username : "";
+                var index = 0, maxTry = 0;
+                var getNextassetSet = function () {
+                    var bamService = Packages.org.wso2.carbon.appmgt.impl.dao.ApiMgtDAO;
+                    var bamServiceObj = new bamService();
+                    log.debug("Object initialized successfully : org.wso2.carbon.appmgt.impl.dao.ApiMgtDAO ");
```
Comment (dinusha-dilrukshi) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: Check isDebugEnabled();

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag [hunk 3]
```diff
@@ -93,7 +93,46 @@ require('/modules/store.js').exec(function (ctx) {
...
+
+                    var result = bamServiceObj.getAppsByHitCount(userName, (paging.start + index), 12, config.uiActivityPublishDataSource);
+                    log.debug("query executed successfully to get the assets hit count");
```
Comment (dinusha-dilrukshi) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: Check isDebugEnabled();

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag [hunk 4]
```diff
@@ -93,7 +93,46 @@ require('/modules/store.js').exec(function (ctx) {
...
+
+                    index += 12;
```
Comment (dinusha-dilrukshi) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: 12 is a magic number. Make it a constant (or a variable)

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag [hunk 5]
```diff
@@ -93,7 +93,46 @@ require('/modules/store.js').exec(function (ctx) {
...
+                    return JSON.parse(result);
+                };
+
+                assets = [];
+                var aid, asset;
```
Comment (dinusha-dilrukshi) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: Use meaningful variable names. "aid"

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag [hunk 6]
```diff
@@ -93,7 +93,46 @@ require('/modules/store.js').exec(function (ctx) {
...
+                while (assets.length < 12 && maxTry < 10) {
```
Comment (dinusha-dilrukshi) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: 10 is a magic number. Make it a constant (or a variable)

File: modules/apps/store/extensions/assets/webapp/pages/assets.jag [hunk 7]
```diff
@@ -96,7 +113,45 @@ require('/modules/store.js').exec(function (ctx) {
                 }
             });
 
-        } else {
+        }
+        else if (paging.sort == 'usage') {
```
Comment (dinusha-dilrukshi) [inline modules/apps/store/extensions/assets/webapp/pages/assets.jag]: Please do the same changes which where mentioned in the same logic of the mobile apps.

File: modules/apps/store/extensions/assets/webapp/themes/store/js/logic/asset/overview/overview.js [hunk 8]
```diff
@@ -1,17 +1,48 @@
...
+        if (isSubscribed.toLowerCase() === 'false') {
+            $('#messageModal2').html($('#confirmation-data2').html());
+            $('#messageModal2 h3.modal-title').html(('Resource forbidden'));
+            $('#messageModal2 div.modal-body').html('\n\n' + ('You have not subscribed to this Application.'));
+            $('#messageModal2 a.btn-other').html('OK');
+
+            $('#messageModal2').modal();
+            e.preventDefault();
+            e.stopPropagation();
+        } else {
```
Comment (dinusha-dilrukshi) [inline modules/apps/store/extensions/assets/webapp/themes/store/js/logic/asset/overview/overview.js]: Remove the else block.

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag (hunk 2 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: Already fixed with a subsequent commit

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag (hunk 3 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: Already fixed with a subsequent commit

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag (hunk 4 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: Already fixed with a subsequent commit

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag (hunk 5 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: Already fixed with a subsequent commit

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag (hunk 6 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: Already fixed with a subsequent commit

File: modules/apps/store/extensions/assets/webapp/pages/assets.jag (hunk 7 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/webapp/pages/assets.jag]: Already fixed with a subsequent commit

File: modules/apps/store/extensions/assets/webapp/themes/store/js/logic/asset/overview/overview.js (hunk 8 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/webapp/themes/store/js/logic/asset/overview/overview.js]: Already fixed with a subsequent commit

File: modules/apps/store/extensions/assets/mobileapp/pages/assets.jag (hunk 1 above)
Comment (lahiru823) [inline modules/apps/store/extensions/assets/mobileapp/pages/assets.jag]: Already fixed with a subsequent commit

**Pull request merged**